src/axom/primal/tests/primal_polygon.cpp

```diff
@@ -718,6 +720,146 @@ TEST(primal_polygon, normal)
   }
 }
 
+template <typename ExecPolicy>
+void check_polygon_policy()
+{
+  const int NUM_VERTS_SQUARE = 4;
+
+  using Polygon3D =
+    axom::primal::Polygon<double, 3, axom::primal::PolygonArray::Static, NUM_VERTS_SQUARE>;
+  using Point3D = axom::primal::Point<double, 3>;
+  using Polygon2D =
+    axom::primal::Polygon<double, 2, axom::primal::PolygonArray::Static, NUM_VERTS_SQUARE>;
+  using Point2D = axom::primal::Point<double, 2>;
+  using Vector3D = axom::primal::Vector<double, 3>;
+
+  Polygon3D* poly_3d_device =
+    axom::allocate<Polygon3D>(1,
+                              axom::execution_space<ExecPolicy>::allocatorID());
+  Polygon2D* poly_2d_device =
+    axom::allocate<Polygon2D>(1,
+                              axom::execution_space<ExecPolicy>::allocatorID());
+
+  Point3D* vertex_mean_3d_device =
+    axom::allocate<Point3D>(1, axom::execution_space<ExecPolicy>::allocatorID());
+  Point2D* vertex_mean_2d_device =
+    axom::allocate<Point2D>(1, axom::execution_space<ExecPolicy>::allocatorID());
+
+  double* area_3d_device =
+    axom::allocate<double>(1, axom::execution_space<ExecPolicy>::allocatorID());
+  double* area_2d_device =
+    axom::allocate<double>(1, axom::execution_space<ExecPolicy>::allocatorID());
+
+  // 3d only
+  Vector3D* normal_3d_device =
+    axom::allocate<Vector3D>(1, axom::execution_space<ExecPolicy>::allocatorID());
+
+  axom::for_all<ExecPolicy>(
+    1,
+    AXOM_LAMBDA(int i) {
+      // Initialize to empty polygons
+      poly_3d_device[i] = Polygon3D();
+      poly_2d_device[i] = Polygon2D();
+      poly_3d_device[i].clear();
+      poly_2d_device[i].clear();
+
+      // Initialize to triangles
+      poly_3d_device[i] = Polygon3D({Point3D({0.0, 0.0, 0.0}),
+                                     Point3D({1.0, 0.0, 0.0}),
+                                     Point3D({1.0, 1.0, 0.0})});
+      poly_2d_device[i] = Polygon2D(
+        {Point2D({0.0, 0.0}), Point2D({1.0, 0.0}), Point2D({1.0, 1.0})});
```

Comment:
Note: This form of initializing a `primal::Polygon` on device was the request in https://github.com/LLNL/axom/issues/1085